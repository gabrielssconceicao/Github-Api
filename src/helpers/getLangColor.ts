export const languageColors: Record<string, string> = {
  JavaScript: '#f7df1e',
  Python: '#3572A5',
  Java: '#b07219',
  'C++': '#f34b7d',
  C: '#555555',
  'C#': '#178600',
  PHP: '#4F5D95',
  TypeScript: '#2b7489',
  Swift: '#ffac45',
  Ruby: '#CC342D',
  Go: '#00ADD8',
  R: '#198CE7',
  Kotlin: '#F18E33',
  Scala: '#c22d40',
  Rust: '#dea584',
  Dart: '#00B4AB',
  Perl: '#0298c3',
  Lua: '#000080',
  Matlab: '#e16737',
  Haskell: '#5e5086',
  SQL: '#e38c00',
  HTML: '#e34c26',
  CSS: '#563d7c',
  'Objective-C': '#438eff',
  Assembly: '#6E4C13',
  Shell: '#89e051',
  PowerShell: '#012456',
  Groovy: '#e69f56',
  Vue: '#41B883',
  Dockerfile: '#384d54',
  Clojure: '#db5855',
  Elixir: '#6e4a7e',
  D: '#ba595e',
  Julia: '#a270ba',
  Arduino: '#bd79d1',
  Verilog: '#b2b7f8',
  VHDL: '#adb2cb',
  Ada: '#02f88c',
  Scheme: '#1e4aec',
  'Objective-C++': '#6866fb',
  Crystal: '#000100',
};
export const getLangColor = (lang: string = ''): string => {
  let language = '#ffffff';
  if (lang in languageColors) {
    language = languageColors[lang];
  }
  return language;
};
