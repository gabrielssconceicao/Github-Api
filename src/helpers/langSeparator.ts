import type { IRepos } from '@/types/repos';
type ILang = { name: string; count: number };
export const langSeparator = (repos: IRepos[]): ILang[] => {
  const langs: ILang[] = [];
  repos.forEach((repo) => {
    const lang = repo.language;
    const index = langs.findIndex((l) => l.name === lang);
    console.log(index);
    if (index === -1) {
      langs.push({ name: lang, count: 1 });
    } else {
      langs[index].count += 1;
    }
    console.log(langs);
  });
  return langs;
};
