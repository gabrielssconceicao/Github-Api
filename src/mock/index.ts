import { type IUser } from '@/types/user';
import { type IRepos } from '@/types/repos';

export const user: IUser = {
  name: 'Teste User',
  login: 'testeuser',
  avatar_url: 'https://avatars.githubusercontent.com/u/10590076?v=4',
  followers: 0,
  following: 5,
  location: 'Brazil',
  html_url: 'https://github.com/testeuser',
  bio: 'lorem ipsum dolor sit amet',
  public_repos: 5,
};
// JS - 3
// HTML - 2
// python - 1
// Ruby - 1
//Java - 1
//C++ - 1
//Typescript - 1
export const repos: IRepos[] = [
  {
    name: 'Teste Repo 1',
    description: 'lorem ipsum dolor sit amet',
    html_url: 'https://github.com/testeuser/testerepo',
    language: 'Javascript',
  },
  {
    name: 'Teste Repo 2',
    description: 'lorem ipsum dolor sit amet',
    html_url: 'https://github.com/testeuser/testerepo',
    language: 'HTML',
  },
  {
    name: 'Teste Repo 3',
    description: 'lorem ipsum dolor sit amet',
    html_url: 'https://github.com/testeuser/testerepo',
    language: 'Python',
  },
  {
    name: 'Teste Repo 4',
    description: 'lorem ipsum dolor sit amet',
    html_url: 'https://github.com/testeuser/testerepo',
    language: 'Javascript',
  },
  {
    name: 'Teste Repo 5',
    description: 'lorem ipsum dolor sit amet',
    html_url: 'https://github.com/testeuser/testerepo',
    language: 'Javascript',
  },
  {
    name: 'Teste Repo 6',
    description: 'lorem ipsum dolor sit amet',
    html_url: 'https://github.com/testeuser/testerepo',
    language: 'HTML',
  },
  {
    name: 'Teste Repo 7',
    description: 'lorem ipsum dolor sit amet',
    html_url: 'https://github.com/testeuser/testerepo',
    language: 'Ruby',
  },
  {
    name: 'Teste Repo 8',
    description: 'lorem ipsum dolor sit amet',
    html_url: 'https://github.com/testeuser/testerepo',
    language: 'Java',
  },
  {
    name: 'Teste Repo 9',
    description: 'lorem ipsum dolor sit amet',
    html_url: 'https://github.com/testeuser/testerepo',
    language: 'C++',
  },
  {
    name: 'Teste Repo 10',
    description: 'lorem ipsum dolor sit amet',
    html_url: 'https://github.com/testeuser/testerepo',
    language: 'Typescript',
  },
];
