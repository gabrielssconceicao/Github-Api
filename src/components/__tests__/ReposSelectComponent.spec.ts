import { describe, it, expect } from 'vitest';
import { mount } from '@vue/test-utils';

import { repos } from '@/mock';
import { langSeparator } from '@/helpers/langSeparator';
import ReposSelect from '../ReposSelectComponent.vue';

describe('<ReposSelect />', () => {
  it('should render with empty repos', () => {
    const wrapper = mount(ReposSelect, {
      props: {
        langs: [],
      },
    });
    expect(wrapper.find('select').exists()).toBe(true);
    expect(wrapper.findAll('option').length).toBe(1);
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render with the langs', () => {
    const langsArray = langSeparator(repos);
    const wrapper = mount(ReposSelect, {
      props: {
        langs: langsArray,
      },
    });
    expect(wrapper.find('select').exists()).toBe(true);
    expect(wrapper.findAll('option').length).toBe(langsArray.length + 1);
    expect(wrapper.element).toMatchSnapshot();
  });
});
