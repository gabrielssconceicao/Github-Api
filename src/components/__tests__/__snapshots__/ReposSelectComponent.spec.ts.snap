// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<ReposSelect /> > should render with empty repos 1`] = `
<select
  class="w-full p-2 bg-primary text-white"
  id="repositories"
  name="repositories"
>
  <option
    value=""
  >
    --All Languages--
  </option>
  
  
</select>
`;

exports[`<ReposSelect /> > should render with the langs 1`] = `
<select
  class="w-full p-2 bg-primary text-white"
  id="repositories"
  name="repositories"
>
  <option
    value=""
  >
    --All Languages--
  </option>
  
  <option
    value="Javascript"
  >
    <span>
      Javascript
    </span>
    <span>
      (3)
    </span>
  </option>
  <option
    value="HTML"
  >
    <span>
      HTML
    </span>
    <span>
      (2)
    </span>
  </option>
  <option
    value="Python"
  >
    <span>
      Python
    </span>
    <span>
      (1)
    </span>
  </option>
  <option
    value="Ruby"
  >
    <span>
      Ruby
    </span>
    <span>
      (1)
    </span>
  </option>
  <option
    value="Java"
  >
    <span>
      Java
    </span>
    <span>
      (1)
    </span>
  </option>
  <option
    value="C++"
  >
    <span>
      C++
    </span>
    <span>
      (1)
    </span>
  </option>
  <option
    value="Typescript"
  >
    <span>
      Typescript
    </span>
    <span>
      (1)
    </span>
  </option>
  
</select>
`;
